src/utilities/effects.ts

```diff
@@ -2580,6 +2581,45 @@ const effects = {
     }
   },
 
+  async getExternalDatasetNames(
+    planId: number,
+    user: User | null,
+    signal: AbortSignal | undefined = undefined,
+  ): Promise<string[]> {
+    try {
+      const data = await reqHasura<PlanDatasetNames[]>(
+        gql.GET_PROFILES_EXTERNAL_NAMES,
+        {
+          planId,
+        },
+        user,
+        signal,
+      );
+      const { plan_dataset: plan_datasets } = data;
+
+      if (plan_datasets != null) {
+        const resourceNames: string[] = [];
+
+        for (const dataset of plan_datasets) {
+          for (const profile of dataset.dataset.profiles) {
+            resourceNames.push(profile.name);
+          }
+        }
+
+        return [...new Set(resourceNames)];
+      } else {
+        throw Error('Unable to get external resource names');
+      }
+    } catch (e) {
+      const error = e as Error;
+      if (error.name !== 'AbortError') {
+        catchError(error);
+        fetchingResourcesExternal.set(false);
```

Comment:
Probably wouldn't want to hook into the main external resource fetching flag? I guess I could be convinced either way. Though i suppose if this query fails I'm not sure why the main one wouldn't also fail.

Reply:
> Probably wouldn't want to hook into the main external resource fetching flag? I guess I could be convinced either way. Though i suppose if this query fails I'm not sure why the main one wouldn't also fail.

I'll make it have it's own flag, this _should_ always complete before fetching the external resources but I'll do it the right way :)